src/main/java/com/revature/responsecollector/JMeterResponseCollector.java

```diff
@@ -0,0 +1,178 @@
+package com.revature.responsecollector;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.jmeter.reporters.ResultCollector;
+import org.apache.jmeter.reporters.Summariser;
+import org.apache.jmeter.samplers.SampleEvent;
+import org.apache.jmeter.samplers.SampleResult;
+
+public class JMeterResponseCollector extends ResultCollector { 
+    
+    private static final long serialVersionUID = 1L;
+    private List<Long> responseTimes = new ArrayList<>();
+    private long responseMax = 0;
+    
+    private long firstSampleStartTime = 0;
+    private long sampleStartTime;
+    
+    // count of number of nXX status codes
+    private int num1XX;
+    private int num2XX;
+    private int num3XX;
+    private int num4XX;
+    private int num5XX;
+    
+    public JMeterResponseCollector(Summariser summer) {
+        super(summer);
+    }
+
+    @Override
+    public void sampleOccurred(SampleEvent e) {
+        super.sampleOccurred(e);
+        SampleResult r = e.getResult();
+        if (firstSampleStartTime == 0) {
+            firstSampleStartTime = r.getStartTime();
+        }
+        sampleStartTime = r.getStartTime();
+        long responseTime = r.getEndTime() - r.getStartTime();
+        responseTimes.add(responseTime);
+        if (responseTime > responseMax) {
+            responseMax = responseTime;
+        }
+        
+        char respType = r.getResponseCode().charAt(0);
+        
+        switch (respType) {
+        case '1':
+            num1XX++;
+            break;
+        case '2':
+            num2XX++;
+            break;
+        case '3':
+            num3XX++;
+            break;
+        case '4':
+            num4XX++;
+            break;
+        case '5':
+            num5XX++;
+            break;
+        default:
+            // TODO log error 
```

Comment:
Once the above field is condensed into a single array, this switch statement can likely be simplified into just incrementing the value at the corresponding index. Will need to convert the String into an Integer however.